frontend/app/src/router/Main.tsx

```diff
@@ -1,17 +1,66 @@
 import { Route, Routes } from 'react-router-dom'
 import Nav from './Nav'
-import { GameView, FriendView, ChatView, ProfileView } from 'view'
+import { GameView, FriendView, ChatView } from 'view'
+import { Profile } from 'components/profile/Profile'
+
+import { mockUser } from 'mock/mockUser'
+import { useEffect, createContext, useState } from 'react'
+import { io, Socket } from 'socket.io-client'
+
+export const SOCKET_EVENT = {
+  JOIN_ROOM: 'JOIN',
+  SEND_MESSAGE: 'SEND',
+  RECEIVE_MESSAGE: 'RECEIVE',
+  NOICE: 'NOTICE',
+  CREATE: 'CREATE',
+}
+
+type Data = {
+  roomId: number
+  senderUid: number
+  msgContent: string
+}
+
+// interface ServerToClientEvents {
+//   noArg: () => void
+//   basicEmit: (a: number, b: string, c: Buffer) => void
+//   withAck: (d: string, callback: (e: number) => void) => void
+// }
+
+// interface ClientToServerEvents {
+//   hello: () => void
+// }
 
 export const MainRouter = () => {
+  const [socket, setSocket] = useState<Socket>()
+  useEffect(() => {
+    const socket = io('/api/chat', {
+      auth: { token: window.localStorage.getItem('access_token') },
+    })
```

Comment:
```suggestion
    const socket: Socket<ServerToClientEvents, ClientToServerEvents> = io('/api/chat', {
      auth: { token: window.localStorage.getItem('access_token') },
    })
```
이렇게 사용하면 enum으로 확인할 필요 없이 각 메소드별로 타입 체크를 받을 수 있어요